ddcommon/src/unix_utils.rs

```diff
@@ -188,17 +188,21 @@ fn is_being_traced() -> io::Result<bool> {
     // Check to see whether we are being traced.  This will fail on systems where procfs is
     // unavailable, but presumably in those systems `ptrace()` is also unavailable.
     // The caller is free to treat a failure as a false.
-    let file = File::open("/proc/self/status")?;
-    let reader = BufReader::new(file);
-
-    for line in reader.lines() {
-        let line = line?;
-        if line.starts_with("TracerPid:") {
-            let tracer_pid = line.split_whitespace().nth(1).unwrap_or("0");
-            return Ok(tracer_pid != "0");
+    // This function may run in signal handler, so we should ensure that we do not allocate
+    // memory on the heap (ex: avoiding using BufReader for exao).
+    let mut file = File::open("/proc/self/status")?;
+    let mut buf = [0u8; 8192];
```

Comment:
One concern that just occurred to me here: this is a fair amount of stack space.  I recall we had issues with stack overflows before: it might be worth using a smaller buffer and looping to avoid that.